state/application.go

```diff
@@ -290,7 +286,6 @@ func (a *Application) removeOps(asserts bson.D) ([]txn.Op, error) {
 		return nil, errors.Trace(err)
 	}
 	ops = append(ops, charmOps...)
-	ops = append(ops, finalAppCharmRemoveOps(name, curl)...)
```

Comment:
finalAppCharmRemoveOps(...) is already called inside appCharmDecRefOps(...) whcih this method calls.

In other words, we add operations from finalAppCharmRemoveOps then we call appCharmDecRefOps which calls finalAppCharmRemoveOps and we add the same operations again :)

We end up with 2 sets of the same operations. Also the correct call is, in fact, appCharmDecRefOps(...) it contains the logic that detects if the unit to be removed is the final unit and adds "final" operations. Keeping this "naked" [1] call to finalAppCharmRemoveOps(...) here, means that we always end up with these operations whether they are needed or not.

[1] not protected by logic - am I final?